cmd/frcli/csv.go

```diff
@@ -27,3 +31,55 @@ func csv(e *frdrpc.ReportEntry) string {
 		e.BtcPrice.Price, e.BtcPrice.PriceTimestamp,
 		e.BtcPrice.Currency, e.Note)
 }
+
+// parsePricesFromCSV reads price point data from the csv at the specified path.
+// This function expects the first csv line to be headers and expects the rest
+// of the lines to be tuples of the following format:
+// 'unix tx seconds, price of 1 BTC in chosen currency'.
+func parsePricesFromCSV(path, currency string) ([]*frdrpc.BitcoinPrice, error) {
+	if path == "" || currency == "" {
+		return nil, errors.New("custom price csv path and " +
+			"currency must both be specified")
+	}
+	csvFile, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer csvFile.Close()
+
+	csvLines, err := csv.NewReader(csvFile).ReadAll()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(csvLines) < 2 {
+		return nil, errors.New("no price points found in CSV")
+	}
+
+	// Skip the first line in the CSV file since we expect this line
+	// to contain column headers.
+	csvLines = csvLines[1:]
+
+	prices := make([]*frdrpc.BitcoinPrice, len(csvLines))
+
+	for i, line := range csvLines {
+		if len(line) < 2 {
+			return nil, errors.New("incorrect csv format. " +
+				"At least two columns items are expected per " +
+				"row")
```

Comment:
Can make this stricter to require exactly 2 columns? Don't want people give weird junk input with extra columns.